app/services/search_indexer.rb

```diff
@@ -53,15 +36,28 @@ def self.update_index(table: , id: , raw_data:)
 
     indexed_data = search_data.select { |d| d.length > 0 }.join(' ')
 
-    params = {
+    ranked_params = {
       a: search_data[0],
       b: search_data[1],
       c: search_data[2],
       d: search_data[3],
+    }
+
+    tsvector = DB.query("SELECT #{ranked_index}", ranked_params)[0].column0
+    additional_lexemes = []
+
+    tsvector.scan(/'(([a-zA-Z0-9]+\.)+[a-zA-Z0-9]+)'\:([\w+,]+)/).reduce(additional_lexemes) do |array, (lexeme, _, positions)|
+      array.concat(lexeme.split('.').map { |l| "'#{l}':#{positions}" })
+    end
```

Comment:
My big question though is, is this injection still too much? 

I my post contains `sam.i.am.hello` should the word `hello` really find this post? Should this injection be specific to domains eg: `https://www.discourse.org`? Even if this is for domains ... should `org` be included as well? what about query params... surely we don't want to inject in query params eg: `https://domain.com?a.b.c.g.e.f.h=1` do we really need h here as a distinct piece?

Reply:
```
discourse_development=# SELECT * FROM TS_DEBUG('https://domain.com?a.b.c.g.e.f.h=1');
  alias   |    description    |     token     | dictionaries | dictionary |     lexemes     
----------+-------------------+---------------+--------------+------------+-----------------
 protocol | Protocol head     | https://      | {}           |            | 
 host     | Host              | domain.com    | {simple}     | simple     | {domain.com}
 blank    | Space symbols     | ?             | {}           |            | 
 file     | File or path name | a.b.c.g.e.f.h | {simple}     | simple     | {a.b.c.g.e.f.h}
 blank    | Space symbols     | =             | {}           |            | 
 uint     | Unsigned integer  | 1             | {simple}     | simple     | {1}
```

Ahh I thought PG was smart enough to only retrieve the host... looks like it treats query params as file here. I wonder if we can identify what type a lexeme is.

Reply:
> If my post contains `sam.i.am.hello` should the word `hello` really find this post?

I think so since this was one of the bug reports we got previously https://meta.discourse.org/t/discourses-internal-search-does-not-find-the-phrase-pagedowncustom-but-google-does/35406/8. 

> what about query params... surely we don't want to inject in query params eg: `https://domain.com?a.b.c.g.e.f.h=1` do we really need h here as a distinct piece?

Query params are tricky and it seems like the PG default parser isn't very smart.

```
discourse_development=# SELECT TO_TSVECTOR('https://meta.discourse.org?test.dot=1');
                to_tsvector                
-------------------------------------------
 '1':3 'meta.discourse.org':1 'test.dot':2
(1 row)

discourse_development=# SELECT TO_TSVECTOR('https://meta.discourse.org/?test.dot=1');
                                to_tsvector                                 
----------------------------------------------------------------------------
 '/?test.dot=1':3 'meta.discourse.org':2 'meta.discourse.org/?test.dot=1':1
(1 row)
```

I guess one thing we can do is to drop query params from the search index?